### PR TITLE
Interactive Toolbar widget

### PR DESCRIPTION
**Pull request by martinRenou**
Make the toolbar an interactive widget.

Now `fig.canvas.toolbar` is a widget, and I also added `fig.canvas.toolbar_visible` and `fig.canvas.toolbar_position` for allowing to hide/move the Toolbar dynamically

### CODE REVIEW

File: ipympl/backend_nbagg.py [hunk 1]
```diff
@@ -21,6 +24,10 @@
 from matplotlib.backend_bases import (ShowBase, NavigationToolbar2,
                                       FigureCanvasBase)
 
+here = os.path.dirname(__file__)
+with open(os.path.join(here, 'static', 'package.json')) as fid:
+    js_version = '^%s' % json.load(fid)['version']
```
Comment (SylvainCorlay) [inline ipympl/backend_nbagg.py]: quick nitpick: maybe `js_semver` instead of `js_version`.

Comment (SylvainCorlay) [reply ipympl/backend_nbagg.py]: (since it is not a version but a semver range)

File: js/src/mpl.js [hunk 2]
```diff
@@ -253,7 +189,21 @@ figure.prototype.handle_rubberband = function(msg) {
 
 figure.prototype.handle_figure_label = function(msg) {
     // Updates the figure title.
-    this.header.textContent = msg['label'];
+    this.figure_label = msg['label'];
+    this.update_header();
```
Comment (SylvainCorlay) [inline js/src/mpl.js]: I think the message info should be in a separate status bar rather than adding text to the title.

Comment (martinRenou) [reply js/src/mpl.js]: It looks better like that indeed

**Comment by martinRenou** [comment]
Thanks! :)

**Pull request merged**